cpp/src/Slice/Parser.cpp

```diff
@@ -803,94 +807,69 @@ Slice::Contained::parseDocComment(function<string(string, string)> linkFormatter
 
     DocCommentPtr comment = make_shared<DocComment>();
```

Comment:
All the remaining code below this is where we parse the actual tags (`@param` etc) into their respective fields on `DocComment`. Before, we had an entire state machine, but this was overkill.

I removed all this logic, and replaced it with a `StringList* currentSection`,
which points to whichever part of the doc-comment we're currently parsing.
At the beginning it points to the overview, but as we encounter tags, we re-point it appropiately.
I find this approach simpler than using a state-machine, and switching on the state to determine where we're writing to.